components/ILIAS/Certificate/classes/Form/Repository/class.ilCertificateSettingsFormRepository.php

```diff
@@ -226,10 +240,20 @@ public function createForm(ilCertificateGUI $certificateGUI): ilPropertyFormGUI
 
         $allowThumbnailDeletion = false;
 
-        $cardThumbnailImagePath = $certificateTemplate->getThumbnailImagePath();
-        if ('' !== $cardThumbnailImagePath) {
-            $presentationThumbnailImagePath = CLIENT_WEB_DIR . $cardThumbnailImagePath;
-            $thumbnailImage->setImage(ilWACSignedPath::signFile($presentationThumbnailImagePath));
+        $thumbnail_image_identification = $certificateTemplate->getThumbnailImageIdentification();
+        $old_thumbnail_image_path = $certificateTemplate->getThumbnailImagePath();
+        if ('' !== $thumbnail_image_identification) {
+            $identification = $this->irss->manage()->find($thumbnail_image_identification);
+            if ($identification instanceof ResourceIdentification) {
+                $thumbnailImage->setImage($this->irss->consume()->src($identification)->getSrc());
+                $allowThumbnailDeletion = true;
+            }
+        } elseif ($this->filesystem->has($old_thumbnail_image_path)) {
+            $thumbnailImage->setImage(
+                ilWACSignedPath::signFile(
+                    ILIAS_HTTP_PATH . '/public/' . ILIAS_WEB_DIR . '/' . CLIENT_ID . $old_thumbnail_image_path
```

Comment:
IMO this is not correct. The `public` folder will not be part of the HTTP path if ILIAS is configured correctly.